Easy Azure Setup.ps1

```diff
@@ -0,0 +1,74 @@
+﻿param(
+$name = "heapkeeper"
+,$resourceGroup = "" # This is used if you want to add this to an existing resource group. if null or empty, the resource group will be set to $name.
+,$location = "centralus" # To see possible locations: az account list-locations -o table
+)
+#A1
+Write-Host "Logging you in, follow the prompts"
+az login | OUT-NULL
+
+if([string]::IsnUllOrEmpty($resourceGroup)){
+    $resourceGroup = $name
+}
+
+
+$getSubscription = $true
+while($getSubscription){
+    try{
+    Write-Host "We are using the following azure subscription, is this correct?:"
+    az account show
+
+        
+    $response = Read-Host "(Y)es/No"
+    if($response -in @("Yes","y")){
+        $getSubscription = $false
+        break
+    }
+        Write-Host "All accounts:"
+        $AllAccounts = az account list | ConvertFrom-Json
+
+        $AllAccounts | select id,name,cloudName, homeTenantID, state | ft
+            
+        $subscription = Read-Host "What subscription should we use? Enter the ID or name"
+
+        az account set --subscription $subscription
+    }
+    catch{
+        Write-Error "There was an error setting the subscription" -ErrorAction Stop
+    }
+}
+#A3
+try{
+$result = az cosmosdb check-name-exists --name $name
+}
+catch{
+    Write-Error "There was an error checking cosmosDB for the name: $name. Ensure you are logged in and that your name does not have underscores"
```

Comment:
@brandonmcclure I did get an error from this once and it was helpful and included the regex they use to validate a name is OK.  Let's output the error message

```suggestion
    Write-Error "There was an error checking cosmosDB for the name: $name. Ensure you are logged in and that your name does not have underscores, the error from the cmdlet follows: $_.Exception.Message"
```